include/xgboost/c_api.h

```diff
@@ -377,7 +377,7 @@ XGB_DLL int XGDMatrixFree(DMatrixHandle handle);
  * \return 0 when success, -1 when failure happens
  */
 XGB_DLL int XGDMatrixSaveBinary(DMatrixHandle handle,
-                                const char *fname, int silent);
+                                const char *fname);
```

Comment:
Please do not change the signature of any C API function, to keep backward compatibility.

Reply:
Oops, I'll revert that. Is it ok to use variable attributes then to mark it as unused?

Reply:
You can make it unnamed, like: `XGB_DLL int XGDMatrixSaveBinary(DMatrixHandle handle, const char *fname, int)`